src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java

```diff
@@ -87,12 +86,14 @@
     private List<ISkinModifier> skinModifiers = Lists.newArrayList();
 
     private SkinResourceManager resources = new SkinResourceManager();
-    //    private ExecutorService executor = Executors.newCachedThreadPool();
 
     private Class<? extends GuiSkins> skinsClass = null;
 
-    private HDSkinManager() {
+    public static PreviewTextureManager getPreviewTextureManager(GameProfile profile) {
```

Comment:
Why is this moved up here? Also, you're undoing a bunch of my work.

Reply:
A bunch of your work here was directly incompatible, and quite frankly unneeded. Preview textures are already loaded with a completable future, it's just not done by wrapping the _entire PreviewTextureManager_.

Reply:
Valhalla needs to do a network call in order to get the locations of the skins. It's also smart to use a future as soon as possible. This is all so the client thread won't lock up.

Reply:
If you're going to change it back, the least you can do is move it back to where it was.

Reply:
I intentionally try to keep the static members near the top of a class where they're not going to get lost.